staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -400,6 +411,19 @@ func (h *UpgradeAwareHandler) tryUpgrade(w http.ResponseWriter, req *http.Reques
 	return true
 }
 
+// this method is taken from net/http/httputil/reverseproxy.go
```

Comment:
@dougsland the function unfortunately isn't exported in https://golang.org/src/net/http/httputil/reverseproxy.go . Maybe I'm missing something, but I don't see a way to use it directly here.

Reply:
> @dougsland the function unfortunately isn't exported in https://golang.org/src/net/http/httputil/reverseproxy.go .
> Maybe I'm missing something, but I don't see a way to use it directly here.

You are correct. I would suggest to share that in the comment too. Additionally (if possible), reach out golang folks to may export the function to avoid duplication code and others projects using it.

For instance:
// FIXME: Taken from net/http/httputil/reverseproxy.go as singleJoiningSlash is not exported to be re-used.  
// See-also: https://github.com/golang/issues/1234567

Reply:
@dougsland changed the comment and opened an issue in the golang repository: https://github.com/golang/go/issues/44290